web/server/entrypoint.go

```diff
@@ -479,11 +495,17 @@ func (s *robotServer) serveWeb(ctx context.Context, cfg *config.Config) (err err
 	}()
 	defer cancel()
 
-	options, err := s.createWebOptions(processedConfig)
+	// As mentioned above, start web service with `minimalProcessedConfig`. Then,
+	// `Reconfigure` robot to have `fullProcessedConfig`.
+	options, err := s.createWebOptions(minimalProcessedConfig)
 	if err != nil {
 		return err
 	}
-	return web.RunWeb(ctx, myRobot, options, s.logger)
+	if err := web.RunWeb(ctx, myRobot, options, s.logger); err != nil {
+		return err
+	}
+	myRobot.Reconfigure(ctx, fullProcessedConfig)
```

Comment:
is there a risk that the config watcher would call reconfigure before this reconfigure is called?

Reply:
Yes; good call out. Discussed offline a bit, we should start the config watcher goroutine only after this reconfigure is called. 

That does mean that we have the same behavior as today: when the robot is starting up (both in first `robotimpl.New(minimalConfig)` and `myRobot.Reconfigure(fullProcessedConfig)`) no new config changes will be seen. So, if a user messes up their config and accidentally starts a module that takes forever to start up, they will not be able to quickly remove that module from their config. Instead, they'll have to restart/shutdown their robot if they want to stop the initial construction. Once again, I don't think this is different from what we have currently, and, of course, `viam-server` is receptive to gRPC requests earlier with the changes in this PR.